File: 240205/줄 세우기 2/line-up-students-2.java
import java.io.*;
import java.util.*;

class Student implements Comparable<Student> {
    int number, height, weight;
    public Student(int number, int height, int weight) {
        this.number = number;
        this.height = height;
        this.weight = weight;
    }
    @Override
    public int compareTo(Student student) {
        if (this.height != student.height) {
            return this.height - student.height;
        }
        return student.weight - this.weight;
    }
    public String toString() {
        return String.format("%d %d %d", this.height, this.weight, this.number);
    }
}

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int inputLength = Integer.parseInt(br.readLine().trim());
        List<Student> students = new ArrayList<>();
        for (int i = 0; i < inputLength; i++) {
            int[] input = Arrays.stream(br.readLine().trim().split(" ")).mapToInt(Integer::parseInt).toArray();
            int number = i + 1;
            int height = input[0];
            int weight = input[1];
            students.add(new Student(number, height, weight));
        }
        Collections.sort(students);
        for (Student student : students) {
            System.out.println(student);
        }
    }
}

File: 240319/줄 세우기 2/line-up-students-2.py
class Person:
    def __init__(self, height, weight, number):
        self.height = height
        self.weight = weight
        self.number = number

    def __str__(self):
        return f'{self.height} {self.weight} {self.number}'
    
    def __repr__(self):
        return self.__str__()

n = int(input())
persons = []
for i in range(n):
    height, weight = list(map(int, input().split()))
    persons.append(Person(height, weight, i+1))

persons.sort(key=lambda ele: (ele.height, -ele.weight))

for ele in persons:
    print(ele)